test/helpers/setup-helper.js

```diff
@@ -52,6 +60,15 @@ process.on('exit', () => {
     process.exit(1);
   }
 });
+// #region Helpers that allow tests to ignore unhandled rejections that might be intentional.
+process.resetIgnoreUnhandled = () => {
+  // default is false
+  ignoreUnhandled = false;
+};
+process.setIgnoreUnhandled = (ignore) => {
+  ignoreUnhandled = ignore;
+};
+// #endregion
```

Comment:
One of the new tests tests that we ignore exceptions, the test passes without the above code addition, but it was annoying to see the `unhandledRejection` output in the logs, so I turned it off.

Note: jest makes it _impossible_ to call `process.removeListener` (or any variation thereof). There is an open issue for it.